### DISCUSSION TITLE
Hardware selection for a computer

### DISCUSSION THREAD

**Discussion by Yason83** [KhronosGroup/KTX-Software#742] [General]
Hello, I'm very grateful for your work.
Currently, we have implemented texture compression and are fully transitioning to this format. 🥇 

Now I'm facing the task of selecting a computer that will perform file conversion.

I have a few questions:

1. Can the conversion be done only on the CPU, or can the graphics card also be utilized?
2. What hardware recommendations do you have?
3. Is there any literature available on this topic that I might have missed?

Thank you.

**Comment by MarkCallow** [comment]
Hmm.

1. KTX-Software uses only the CPU for compressing images. It should be possible to write a GPU compute shader that does it. I'm not aware of any such effort. If compressing to one of the Basis Universal formats (basis-lz or UASTC) and running on an x64 processor, SSE SIMD instructions can be used. If compressing to ASTC, either SSE or AVX2 can be used.  If running on an arm64 processor, only ASTC encoding can use the Neon SIMD instructions.,  Whether or not they are used is controlled by the `--threads` option of `ktx create`, `ktx encode`, `toktx` and `ktxsc`. Make sure your software is compiled for Release mode. Encoding in debug mode is slow.
2. I don't have any hardware recommendations. Generally any fast processor will do. If you will be processing very large images then decent sized main memory will help. If your focus is basis-lz or UASTC then get an x64 processor so you can use SIMD though my M2 MacBook is no slouch even without SIMD.
3. Not that I am aware of.

**Reply by Yason83** [reply]
Thaks for your recomendation
